tests/test_spider.py

```diff
@@ -360,10 +362,12 @@ def process_request_upper(self, request, response):
         output = list(spider._requests_to_follow(response))
         self.assertEqual(len(output), 3)
         self.assertTrue(all(map(lambda r: isinstance(r, Request), output)))
-        self.assertEqual([r.url for r in output],
-                         ['http://EXAMPLE.ORG/SOMEPAGE/ITEM/12.HTML',
-                          'http://EXAMPLE.ORG/ABOUT.HTML',
-                          'http://EXAMPLE.ORG/NOFOLLOW.HTML'])
+        urls = ['http://EXAMPLE.ORG/SOMEPAGE/ITEM/12.HTML',
+                'http://EXAMPLE.ORG/ABOUT.HTML',
+                'http://EXAMPLE.ORG/NOFOLLOW.HTML']
+        if parse_version(w3lib_version) >= parse_version('2.0.0'):
+            urls = list(map(lambda u: u.replace("EXAMPLE.ORG", "example.org"), urls))
+        self.assertEqual([r.url for r in output], urls)
```

Comment:
I ask as someone completely unfamiliar with this test: would it make sense to change the test so that valid domains are used? Or is the use of invalid domains page of what the test is checking for? And, if the later, why? Should we also skip this test with w3lib 2+ and plan to think of a way to test this on w3lib 2+ later?

Reply:
I've looked at the test and thought that it just tests that the `process_request_upper()` method actually ran, by checking that the URLs were modified. Assuming this is true, it can be changed to some other modification.

Reply:
@wRAR @Gallaecio  Maybe we can use `safe_url_string` for each url, with this the output will be standarized for all `w3lib` versions. What you guys think?

Reply:
https://github.com/scrapy/scrapy/pull/5617/commits/582a6bf6dbcb01da40cbec6b51269add2db39cf1 is not what I thought you meant, I thought you meant using safe_url_string in the method that uppercases the URLs originally.

That said, it seems like a very valid alternative to me. Great job!

Reply:
@wRAR Feel free to merge if you are also OK with the change.